backend/setup.js

```diff
@@ -5,111 +5,145 @@ const logger              = require('./logger').setup;
+				nickname:   'Admin',
+				avatar:     '',
+				roles:      ['admin']
+			};
+
 			return userModel
 				.query()
-				.select(userModel.raw('COUNT(`id`) as `count`'))
-				.where('is_deleted', 0)
-				.first();
-		})
-		.then((row) => {
-			if (!row.count) {
-				// Create a new user and set password
-				logger.info('Creating a new user: admin@example.com with password: changeme');
-
-				let data = {
-					is_deleted: 0,
-					email:      'admin@example.com',
-					name:       'Administrator',
-					nickname:   'Admin',
-					avatar:     '',
-					roles:      ['admin']
-				};
+				.insertAndFetch(data)
+				.then( (user) => {
+					return authModel
+						.query()
+						.insert({
+							user_id: user.id,
+							type:    'password',
+							secret:  'changeme',
+							meta:    {}
+						})
+						.then(() => {
+							return userPermissionModel
+								.query()
+								.insert({
+									user_id:           user.id,
+									visibility:        'all',
+									proxy_hosts:       'manage',
+									redirection_hosts: 'manage',
+									dead_hosts:        'manage',
+									streams:           'manage',
+									access_lists:      'manage',
+									certificates:      'manage'
+								});
+						});
+				})
+				.then(() => {
+					logger.info('Initial admin setup completed');
+				});
+		} else if (debug_mode) {
+			logger.debug('Admin user setup not required');
+		}
+	});
+}
 
-				return userModel
+function setupDefaultSettings() {
+	return settingModel
+		.query()
+		.select(userModel.raw('COUNT(`id`) as `count`'))
+		.first()
+		.then( (row) => {
+			if (!row.count) {
+				settingModel
 					.query()
-					.insertAndFetch(data)
-					.then((user) => {
-						return authModel
-							.query()
-							.insert({
-								user_id: user.id,
-								type:    'password',
-								secret:  'changeme',
-								meta:    {}
-							})
-							.then(() => {
-								return userPermissionModel
-									.query()
-									.insert({
-										user_id:           user.id,
-										visibility:        'all',
-										proxy_hosts:       'manage',
-										redirection_hosts: 'manage',
-										dead_hosts:        'manage',
-										streams:           'manage',
-										access_lists:      'manage',
-										certificates:      'manage'
-									});
-							});
-					})
-					.then(() => {
-						logger.info('Initial setup completed');
+					.insert({
+						id:          'default-site',
+						name:        'Default Site',
+						description: 'What to show when Nginx is hit with an unknown Host',
+						value:       'congratulations',
+						meta:        {}
+					}).then(() => {
+						logger.info('Default settings added');
 					});
-			} else if (debug_mode) {
-				logger.debug('Admin user setup not required');
+			} if (debug_mode) {
+				logger.debug('Default setting setup not required');
 			}
 		});
+}
+
+module.exports = function () {
+	return new Promise((resolve, reject) => {
+		return setupJwt(resolve, reject);
```

Comment:
setupJwt is not a promise, so returning it as is will be async but won't reject on error properly. If it were a promise, you wouldn't need to wrap it in a Promise constructor either.

Reply:
I literally just extracted the code to a new function. I think if it has some logical error now, it had it in the past too.

Probably I could have moved the promise creation to the `setupJwt` too, but probably it is just a matter of taste.

Reply:
ohhh, or I just simply not need to write the "return" keyword in this case?

Reply:
Not really, it just could be written differently but let's just get it through for now